### PR TITLE
Added Endpoints for Prescription/MedicationRequest Resource

### PR DESCRIPTION
**Pull request by yashrajbothra**
Fixes #

#### Changes proposed in this pull request:

- Added Prescription Endpoints
- Added MedicationRequest Endpoints
- Changed Drug Code to support only RXCUI Code

### CODE REVIEW

**Comment by yashrajbothra** [comment]
https://github.com/openemr/openemr/blob/c9825e8b969c0eb4786b7b5de897b0908fc9b756/interface/patient_file/encounter/find_code_dynamic.php#L154
The callback in [find_code_dynamic](https://github.com/openemr/openemr/blob/c9825e8b969c0eb4786b7b5de897b0908fc9b756/interface/patient_file/encounter/find_code_dynamic.php#L118) doesn't support target. Should we add ```sel_related_target``` like this https://github.com/openemr/openemr/blob/932ade32bca6b28541d5b851b2d79638b3a1e94a/interface/patient_file/encounter/find_code_popup.php#L88

File: sql/5_0_2-to-6_0_0_upgrade.sql [hunk 1]
```diff
@@ -1984,3 +1984,18 @@ ALTER TABLE `drugs` ADD `uuid` binary(16) DEFAULT NULL;
...
+
+#IfUuidNeedUpdate prescriptions
+#EndIf
+
+#IfNotIndex prescriptions uuid
+CREATE UNIQUE INDEX `uuid` ON `prescriptions` (`uuid`);
+#EndIf
+
+#IfNotColumnType prescriptions rxnorm_drugcode varchar(25)
+ALTER TABLE `prescriptions` MODIFY `rxnorm_drugcode` varchar(25) DEAFULT NULL;
```
Comment (bradymiller) [inline sql/5_0_2-to-6_0_0_upgrade.sql:2000]: mispell DEFAULT

File: sql/database.sql [hunk 2]
```diff
@@ -7224,7 +7225,7 @@ CREATE TABLE `prescriptions` (
   `start_date` date default NULL,
   `drug` varchar(150) default NULL,
   `drug_id` int(11) NOT NULL default '0',
-  `rxnorm_drugcode` INT(11) DEFAULT NULL,
+  `rxnorm_drugcode` varchar(25) DEAFULT NULL,
```
Comment (bradymiller) [inline sql/database.sql:7228]: mispell DEFAULT

**Comment by bradymiller** [comment]
travis is pointing out a php deprecated error for 7.4 and 8.0:
```
------------------------------
Checking for PHP syntax errors
------------------------------
PHP Deprecated:  Cannot use "parent" when current class scope has no parent in ./src/Services/FHIR/FhirMedicationRequestService.php on line 28
---------------------------------------
```

File: src/Services/PrescriptionService.php [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                drug.uuid AS drug_uuid
+                FROM prescriptions
+                LEFT JOIN patient_data AS patient
+                ON patient.pid = prescriptions.patient_id
+                LEFT JOIN form_encounter AS encounter
+                ON encounter.encounter = prescriptions.encounter
+                LEFT JOIN users AS practitioner
+                ON practitioner.id = prescriptions.provider_id
+                LEFT JOIN drugs AS drug
+                ON drug.drug_id = prescriptions.drug_id";
```
Comment (bradymiller) [inline src/Services/PrescriptionService.php:62]: Pretty sure all the potential uuid's in here need to be populated in __contruct , correct?

**Comment by bradymiller** [comment]
hi @yashrajbothra , Just let me know when this is ready for final testing before bringing into codebase. thanks! -brady

**Comment by yashrajbothra** [comment]
Hii @bradymiller , All Endpoints are testing well. But the frontend part is not testing well because of the callback ```sel_rel()``` . The data after selecting code are adding to **related to** although it should be added to **RXCUI Code** I think we need to change callback to ```sel_rel_target```. Also how can we limit the modal to select only one code?

**Comment by yashrajbothra** [comment]
Hello @bradymiller Sir, Pushed a fix for Target Element and Added Limit in the Popup. Kindly Review :)

Thanks :)

**Comment by bradymiller** [comment]
hi @yashrajbothra ui and rxnorm stuff is testing well. I'm assuming the api endpoints are testing well for you, so bringing this in. thanks for the FHIR endpoints! -brady

**Pull request merged**